src/lua/utils.lua

```diff
@@ -111,4 +114,34 @@ function utils.update_param_table(table, defaults)
     return new_table
 end
 
+ffi.cdef[[
+void
+__asan_poison_memory_region(void const volatile *addr, size_t size);
+void
+__asan_unpoison_memory_region(void const volatile *addr, size_t size);
+int
+__asan_address_is_poisoned(void const volatile *addr);
+]]
+
+if tarantool.build.asan then
+    utils.poison_memory_region = function(start, size)
+        ffi.C.__asan_poison_memory_region(start, size)
+    end
+    utils.unpoison_memory_region = function(start, size)
+        ffi.C.__asan_unpoison_memory_region(start, size)
+    end
+    utils.region_is_poisoned = function(start, size)
+        for i = 0, size - 1 do
+            if not ffi.C.__asan_address_is_poisoned(start + i) then
```

Comment:
This function returns int (not bool). `not 0` is false. So this function will always return true.

Reply:
Thanx! Fixed.

```diff
diff --git a/src/lua/utils.lua b/src/lua/utils.lua
index 85dc9fc5b..cc1cf5643 100644
--- a/src/lua/utils.lua
+++ b/src/lua/utils.lua
@@ -132,7 +132,7 @@ if tarantool.build.asan then
     end
     utils.region_is_poisoned = function(start, size)
         for i = 0, size - 1 do
-            if not ffi.C.__asan_address_is_poisoned(start + i) then
+            if ffi.C.__asan_address_is_poisoned(start + i) == 0 then
                 return false
             end
         end
```